### PR TITLE
Support configurable user interface components

### PR DESCRIPTION
**Pull request by pepermao**
image
image
image

### CODE REVIEW

File: src/components/Footer/Aletheiainfoadress.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { useTranslation } from "next-i18next";
+import { useAppSelector } from "../../store/store";
+
+const Aletheiainfoadress = () => {
```
Comment (pepermao) [inline src/components/Footer/Aletheiainfoadress.tsx:4]: Suggestion (non-blocking): Utilize the Camel Case pattern `AletheiaInfoAdress`

File: src/components/Footer/Aletheiainfoadress.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  const { t } = useTranslation();
+  const { vw } = useAppSelector((state) => state);
+
+  return (
+    <>
+      <span
+        style={{
+          fontSize: "14px",
+          marginTop: "0px",
+          textAlign: "center",
```
Comment (pepermao) [inline src/components/Footer/Aletheiainfoadress.tsx:14]: Suggestion: Create a styled component using the `styled-component` library allowing code reuse.

File: src/components/Footer/AletheiaSocialMediaFooter.tsx [hunk 3]
```diff
@@ -24,50 +26,20 @@ const AletheiaSocialMediaFooter = () => {
                 </h3>
             </Col>
             <Col span={24}>
-                <SocialIcon
```
Comment (pepermao) [inline src/components/Footer/AletheiaSocialMediaFooter.tsx:27]: chore: Create a React component called `AletheiaSocialMediaIcons.tsx` to encapsulate the social media icons code

File: src/components/Footer/AletheiaSocialMediaFooter.tsx [hunk 4]
```diff
@@ -24,50 +26,20 @@ const AletheiaSocialMediaFooter = () => {
...
-                    bgColor={
-                        nameSpace === NameSpaceEnum.Main
-                            ? colors.bluePrimary
-                            : colors.blueSecondary
-                    }
-                    target="_blank"
-                    rel="noreferrer"
-                    fgColor="white"
-                />
+                {localConfig.footer.socialMedias.map((url, index) => (
```
Comment (pepermao) [inline src/components/Footer/AletheiaSocialMediaFooter.tsx:29]: chore: Add conditional checking if the socialMedias array is empty

File: src/components/Header/Logo.tsx [hunk 5]
```diff
@@ -11,29 +11,14 @@ const Logo = ({
     lineHeight = "24px",
 }) => {
     const [nameSpace] = useAtom<string>(currentNameSpace);
-
+    const logoSvgPath = process.env.LOGO_SVG_PATH || '/images/logo1_white.svg';
     if (nameSpace === NameSpaceEnum.Main) {
         return (
-            <svg
-                data-cy={"logo"}
+            <img
```
Comment (thesocialdev) [inline src/components/Header/Logo.tsx:17]: We should use NextJS image component instead.

File: src/components/Header/DonateButton.tsx [hunk 6]
```diff
@@ -20,7 +21,7 @@ const DonateButton = ({ header = false, style = {} }) => {
                 ...style,
             }}
         >
-            {header ? t("header:donateButton") : t("home:donateButton")}
+            {t("home:donateButton")}
```
Comment (thesocialdev) [inline src/components/Header/DonateButton.tsx:24]: question: any reason for removing this logic?

File: config/localConfig.ts [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        affixCTA: z.boolean(),
+    }),
+});
+
+type LocalConfig = z.infer<typeof LocalConfigSchema>;
+
+const localConfig: LocalConfig = {
+    theme: {
+        colors: {
+            bluePrimary: "#11273a",
```
Comment (thesocialdev) [inline config/localConfig.ts:55]: issue: the template shouldn't have the colors here. Aletheia default theme should remain in the colors object

**Review by thesocialdev — CHANGES_REQUESTED** [review]

File: src/components/Header/DonateButton.tsx (hunk 6 above)
Comment (LuizFNJ) [reply src/components/Header/DonateButton.tsx:24]: We were configuring the text of this button in the local config. After switching to configuring by overwriting the JSONs, I believe this logic is no longer necessary.

File: src/components/Header/DonateButton.tsx [hunk 8]
```diff
@@ -2,13 +2,14 @@ import React from "react";
...
 import { useTranslation } from "next-i18next";
+import localConfig from "../../../config/localConfig.example";
 
 const DonateButton = ({ header = false, style = {} }) => {
     const { t } = useTranslation();
     return (
         <AletheiaButton
             type={ButtonType.white}
-            href="https://donate.aletheiafact.org/"
+            href={localConfig.header.donateButton.redirectUrl ? localConfig.header.donateButton.redirectUrl : "https://donate.aletheiafact.org/"}
```
Comment (thesocialdev) [inline src/components/Header/DonateButton.tsx:12]: suggestion: for the future we should just get this string from the locale JSONs

File: src/components/Footer/FooterInfo.tsx [hunk 9]
```diff
@@ -6,38 +6,40 @@ import { useAppSelector } from "../../store/store";
 
 import colors from "../../styles/colors";
 import AletheiaButton, { ButtonType } from "../Button";
+import localConfig from "../../../config/localConfig.example";
 
 const FooterInfo = () => {
     const { t } = useTranslation();
     const { vw } = useAppSelector((state) => state);
+    const href = localConfig.footer.repositoryLink ? localConfig.footer.repositoryLink : " https://github.com/AletheiaFact/aletheia"
```
Comment (thesocialdev) [inline src/components/Footer/FooterInfo.tsx:14]: issue: this should always be Aletheia's original repo URL

File: src/components/Header/Logo.tsx [hunk 10]
```diff
@@ -4,36 +4,24 @@ import colors from "../../styles/colors";
...
-                    fill={color}
-                />
-                <path
-                    d="M957 76.024v337.27a2.096 2.096 0 0 1-.632 1.515c-.41.405-.971.637-1.558.645h-62.461c-.477 0-.933-.183-1.27-.507a1.693 1.693 0 0 1-.526-1.221v-20.944c0-.458.189-.898.526-1.222a1.832 1.832 0 0 1 1.27-.506h38.601V98.264H26.057v292.79h424.151a2.26 2.26 0 0 1 1.562.625c.205.198.367.432.478.691.11.258.166.535.165.814v20.11c.003.281-.051.56-.161.821a2.118 2.118 0 0 1-.476.699 2.212 2.212 0 0 1-.718.47 2.29 2.29 0 0 1-.85.17H2.19a2.28 2.28 0 0 1-.848-.17 2.211 2.211 0 0 1-.715-.471 2.104 2.104 0 0 1-.471-.699 2.053 2.053 0 0 1-.156-.82V76.024c0-.573.237-1.122.658-1.527a2.29 2.29 0 0 1 1.587-.633H954.81a2.27 2.27 0 0 1 1.556.647c.41.404.638.948.634 1.513Z"
-                    fill={color}
```
Comment (thesocialdev) [inline src/components/Header/Logo.tsx:30]: issue: this will break the logic of changing the SVG color

**Comment by thesocialdev** [comment]
@LuizFNJ I'm going to merge this so I can start working on the deployment step, but please fix the comments I made on a separate branch and PR.

File: src/components/Header/Logo.tsx [hunk 11]
```diff
@@ -4,36 +4,24 @@ import colors from "../../styles/colors";
...
 
 const Logo = ({
     color = colors.logoWhite,
     height = "42px",
+    width = "80px",
     lineHeight = "24px",
 }) => {
     const [nameSpace] = useAtom<string>(currentNameSpace);
-
+    const logoSvgPath = process.env.LOGO_SVG_PATH || '/images/logo1_white.svg';
```
Comment (thesocialdev) [inline src/components/Header/Logo.tsx:16]: issue: this is the only place using process.env instead of localConfig

**Pull request merged**